src/storage/src/hummock/sstable_store.rs

```diff
@@ -254,6 +260,54 @@ impl SstableStore {
             .map_err(HummockError::object_io_error)
     }
 
+    pub async fn preload_blocks(
+        &self,
+        sst: &Sstable,
+        mut start_index: usize,
+        mut end_index: usize,
+    ) -> HummockResult<Option<BatchBlockStream>> {
+        let object_id = sst.id;
+        while start_index < end_index
+            && self.block_cache.exists_block(object_id, start_index as u64)
+        {
+            start_index += 1;
+        }
+        if start_index >= end_index {
+            return Ok(None);
+        }
+        let start_offset = sst.meta.block_metas[start_index].offset as usize;
+        let mut end_offset = start_offset;
+        for idx in start_index..end_index {
+            if self.block_cache.exists_block(object_id, idx as u64) {
+                end_index = idx;
+                break;
+            }
+            end_offset += sst.meta.block_metas[idx].len as usize;
+        }
+        let pending_data = self.pending_streaming_loading.load(Ordering::SeqCst);
+        if end_offset == start_offset || pending_data > self.large_query_memory_usage {
+            return Ok(None);
+        }
+        let data_path = self.get_sst_data_path(object_id);
+        let memory_usage = std::cmp::min(end_offset - start_offset, self.store.recv_buffer_size());
+        self.pending_streaming_loading
+            .fetch_add(memory_usage, Ordering::SeqCst);
+        let reader = self
```

Comment:
I think `large_query_memory_usage` and `pending_streamin_loading` default parameters are too pessimistic, because I ran a benchmark on our cluster which shows that even if we don't limit any streaming read concurrency, we can maximize the S3 throughput and maintain low memory consumption.

Reply:
Without limitations of concurrent streamin_read:

We can maximize the S3 throughput
https://files.slack.com/files-pri/T030LTU38S2-F0656N1EJR1/image.png

and maintain low memory consumption.
![image](https://github.com/risingwavelabs/risingwave/assets/9352536/48419084-e226-4579-9260-482cdb88f50f)

Reply:
While under current limitations, the streaming read will degrade to read.
https://files.slack.com/files-pri/T030LTU38S2-F06507KJ87Q/image.png

Reply:
Thanks for your benchmark testing. I have set `large_query_memory_usage ` default equal with `block_cache_size`. And the `low_priority_cache_ratio` for large query has increased from 30% to 50%.